src/device.rs

```diff
@@ -316,7 +316,18 @@ impl Device {
                 "Couldn't find an encoder for that connector.",
             ))?;
 
-        let crtc = encoder.crtcs().into_iter().next().ok_or(io::Error::new(
+        let crtc_id = connector
+            .property("CRTC_ID")?
+            .map(|id| id.value())
+            .and_then(|id| u32::try_from(id).ok())
+            .filter(|id| *id > 0);
+
+        let crtc = if let Some(id) = crtc_id {
+            encoder.crtc_with_id(id)
+        } else {
+            encoder.crtcs().into_iter().next()
+        }
```

Comment:
I think we can simplify / functional program it further with:

```rust
        let crtc = connector
            .property("CRTC_ID")?
            .map(|id| u32::try_from(id.value()).expect("We cannot have more than 32 CRTCs"))
            .filter(|id| *id > 0)
            .map_or_else(
                || encoder.crtcs().into_iter().next(),
                |id| encoder.crtc_with_id(id),
            )
            .ok_or(io::Error::new(
                io::ErrorKind::NotFound,
                "Couldn't find a CRTC for that connector.",
            ))?;
```
WDYT?

Reply:
Good idea! I'll leave the `u32::try_from(id).ok()` because `expect()` triggers this warning:

```
used unwrap or expect in a function that returns result or option
unwrap and expect should not be used in a function that returns result or option
for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unwrap_in_result
requested on the command line with `-W clippy::unwrap-in-result`
```

Reply:
I still think we need an expect here. There's no reason a CRTC ID wouldn't fit in a u32 except if something seriously broken is happening. panic'ing is the best we can do in that situation.

You can disable the clippy warning using

```
    #[expect(
        clippy::unwrap_in_result,
        reason = "There's no way a CRTC ID doesn't fit into a u32."
    )]
```

Right before the function prototype

Reply:
Done, thanks!